### PR TITLE
UIOR-1078: Allow Editing of "Renewal Date" and "Subscription to" on o…

### PR DESCRIPTION
**Pull request by alisher-epam**
…pen purchase order line

## Purpose

[UIOR-1078](https://issues.folio.org/browse/UIOR-1078) - Allow Editing of "Renewal Date" and "Subscription to" on open purchase order line
## Approach

https://github.com/folio-org/ui-orders/assets/116072773/6e347735-2f4e-4f29-9c3d-eefd0a5e4de2

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/PurchaseOrder/OngoingOrderInfo/OngoingInfoForm.js [hunk 1]
```diff
@@ -51,7 +52,7 @@ const OngoingInfoForm = ({ hiddenFields = {} }) => {
             <VisibilityControl name="hiddenFields.ongoing.isSubscription">
               <FieldRenewalSubscription
                 disabled={disabled}
-                isNonInteractive={isNonPendingNonInteractive}
+                isNonInteractive={isNonPendingNonInteractive && !isWorkflowStatusOpen(values)}
```
Comment (mariia-aloshyna) [inline src/components/PurchaseOrder/OngoingOrderInfo/OngoingInfoForm.js:55]: Move `!isWorkflowStatusOpen(values)` to a variable, please

**Review by BogdanDenis — APPROVED** [review]

**Comment by usavkov-epam** [comment]
Just in case: this story is linked to "Q" release, so please do NOT merge it until the release of `ui-orders`

**Review by mariia-aloshyna — APPROVED** [review]

**Comment by alisher-epam** [comment]
> I didn’t really understand this PR. The "Purpose" of your PR and the description of the ticket [UIOR-1078](https://issues.folio.org/browse/UIOR-1078) tell about changes in the PO line form, but I see in this PR work on the PO form.

Yeah, basically in order to get Open status PR it should have PO lines and we can create it from action menus
Screenshot 2023-10-13 at 8 12 19 PM

**Comment by usavkov-epam** [comment]
Having merged this PR, are you sure that it fully complies with the requirements of UIOR-1078?

**Comment by usavkov-epam** [comment]
And don't we need to bump the version to `5.1.0` in the `package.json` beforehand? @zburke

**Comment by zburke** [comment]
Yes, `package.json` needs to be bumped to `5.1.0`. I've tried to get in the habit of doing that on the master branch as soon as I make a release, but many folks just do it in the first new-feature PR. Up to you, as long as it happens :)

**Pull request merged**